apis/duck/v1/kresource_type.go

```diff
@@ -33,6 +33,8 @@ type KRShaped interface {
 	GetTypeMeta() *metav1.TypeMeta
 
 	GetStatus() *Status
+
+	GetTopLevelConditionType() apis.ConditionType
```

Comment:
Too late, but I'm curious why we didn't stick to the current naming with here (`Happy` instead of `TopLevel`).